utils/yoke/src/yoke.rs

```diff
@@ -116,12 +120,18 @@ impl<Y: for<'a> Yokeable<'a>, C: StableDeref> Yoke<Y, C> {
     /// ```
     pub fn attach_to_cart<F>(cart: C, f: F) -> Self
     where
+        // safety note: This works by enforcing that the *only* place the return value of F
+        // can borrow from is the cart, since `F` must be valid for all lifetimes `'de`
+        //
+        // safety note 2: this is partially a lie due to the implied bound `C: 'de`,
```

Comment:
The implied bound isn't `C: 'de`, it's `C::Target: 'de` as elaborated later. Being incorrect here mislead me temporarily as to the nature of the problem (thinking it was due to `C: 'de` offering a hole `C::Target: 'de` wouldn't).

Reply:
yeah sorry, I had reduced the testcase to file upstream and had removed the deref, and I wasn't fully careful about getting back to the correct terms when here